src/main/java/org/jenkinsci/plugins/workflow/cps/view/InterpolatedSecretsAction.java

```diff
@@ -0,0 +1,193 @@
+                            }
+                        } else {
+                            failReason = "null flow node";
+                        }
+                    } else {
+                        failReason = "null flow execution owner";
+                    }
+                } catch (IOException e) {
+                    failReason = "could not get flow node";
+                }
+            } else {
+                failReason = "not an instance of FlowExecutionOwner.Executable";
+            }
+            throw new IllegalStateException(failReason);
+        }
+
+        @Exported
+        public List<String> getInterpolatedVariables() {
+            return interpolatedVariables;
+        }
+    }
+
+    private static String argumentToString(Object arg) {
+        String valueString;
+        if (arg instanceof Map.Entry) {
+            Map.Entry argEntry = (Map.Entry<String, Object>) arg;
+            Object value = argEntry.getValue();
+            if (value instanceof ArgumentsAction.NotStoredReason) {
+                switch ((ArgumentsAction.NotStoredReason) value) {
+                    case OVERSIZE_VALUE:
+                        valueString = "argument omitted due to length";
+                        break;
+                    case UNSERIALIZABLE:
+                        valueString = "unable to serialize argument";
+                        break;
+                    default:
+                        valueString = String.valueOf(value);
+                        break;
+                }
+            } else if (value instanceof Map || value  instanceof List || value instanceof UninstantiatedDescribable) {
+                valueString = argumentToString(value);
+            } else {
+                valueString = String.valueOf(value);
+            }
+            return argEntry.getKey() + ": " + valueString;
+        } else if (arg instanceof Map) {
+            valueString = ((Map<?, ?>) arg).entrySet().stream()
+                    .map(InterpolatedSecretsAction::argumentToString)
+                    .collect(Collectors.joining(", ", "[", "]"));
+        } else if (arg instanceof List) {
+            valueString = ((List<?>) arg).stream()
+                    .map(InterpolatedSecretsAction::argumentToString)
+                    .collect(Collectors.joining(", ", "[", "]"));
+        } else if (arg instanceof UninstantiatedDescribable) {
+            UninstantiatedDescribable ud = (UninstantiatedDescribable) arg;
+            String udName = null;
+            if (ud.getSymbol() != null) {
+                udName = '@' + ud.getSymbol();
+            }
+            if (ud.getKlass() != null) {
+                udName = '$' + ud.getKlass() + udName;
+            }
+
+            Map<String, ?> udArgs = ud.getArguments();
+            valueString = udArgs.entrySet().stream()
+                    .map(InterpolatedSecretsAction::argumentToString)
+                    .collect(Collectors.joining(", ", udName + "(", ")"));
```

Comment:
This is right when using symbols, but when `$class` is used the syntax is a list like `[$class: udName, args...]`.

Reply:
(A map, not a list.)

Reply:
updated in b52438dc7576b6620015f24d08baa6990b6fa120 to handle `$class`. Not sure what I should be using to test those changes. I just looked up some examples I found online.

Reply:
Basically any extension without a `@Symbol`.